packages/admin-ui/ui/src/domain/products/edit/sections/variants/edit-variant-inventory-modal.tsx

```diff
@@ -0,0 +1,192 @@
+import { Product, ProductVariant } from "@medusajs/medusa"
+import {
+  useAdminUpdateLocationLevel,
+  useAdminVariantsInventory,
+} from "medusa-react"
+import React, { useContext } from "react"
+import { useForm } from "react-hook-form"
+import Button from "../../../../../components/fundamentals/button"
+import Modal from "../../../../../components/molecules/modal"
+import LayeredModal, {
+  LayeredModalContext,
+} from "../../../../../components/molecules/modal/layered-modal"
+import EditFlowVariantForm, {
+  EditFlowVariantFormType,
+} from "../../../components/variant-inventory-form/edit-flow-variant-form"
+import useEditProductActions from "../../hooks/use-edit-product-actions"
+import { createUpdatePayload } from "./edit-variants-modal/edit-variant-screen"
+
+type Props = {
+  onClose: () => void
+  product: Product
+  variant: ProductVariant
+  isDuplicate?: boolean
+}
+
+const EditVariantInventoryModal = ({ onClose, product, variant }: Props) => {
+  const layeredModalContext = useContext(LayeredModalContext)
+  const {
+    // @ts-ignore
+    variant: variantInventory,
+    isLoading: isLoadingInventory,
+    refetch,
+  } = useAdminVariantsInventory(variant.id)
+
+  const itemId = variantInventory?.inventory[0]?.id
+
+  const { mutate: updateLocationLevel } = useAdminUpdateLocationLevel(
+    itemId || ""
+  )
+  const handleClose = () => {
+    onClose()
+  }
+
+  const { onUpdateVariant, updatingVariant } = useEditProductActions(product.id)
+
+  const onSubmit = async (data) => {
+    const { location_levels } = data.stock
+
+    await Promise.all(
+      location_levels.map(async (level) => {
+        await updateLocationLevel({
+          stockLocationId: level.location_id,
+          stocked_quantity: level.stocked_quantity,
+        })
+      })
+    )
+    // / TODO: Call update location level with new values
+    delete data.stock.location_levels
+
+    // @ts-ignore
+    onUpdateVariant(variant.id, createUpdatePayload(data), () => {
+      refetch()
+      handleClose()
+    })
+  }
+
+  return (
+    <LayeredModal context={layeredModalContext} handleClose={handleClose}>
+      <Modal.Header handleClose={handleClose}>
+        <h1 className="inter-xlarge-semibold">Edit stock & inventory</h1>
+      </Modal.Header>
+      {!isLoadingInventory && (
+        <StockForm
+          variantInventory={variantInventory}
+          refetchInventory={refetch}
+          onSubmit={onSubmit}
+          isLoadingInventory={isLoadingInventory}
+          handleClose={handleClose}
+          updatingVariant={updatingVariant}
+        />
+      )}
+    </LayeredModal>
+  )
+}
+
+const StockForm = ({
```

Comment:
todo: we get an exception if the inventory module is not installed, should we have a fallback